app/models/schedule.rb

```diff
@@ -137,4 +137,28 @@ def clean_up_task_jobs
     end
   end
 
+  def timeline
+    timeline = []
+    self.root_task.self_and_descendants.each do |task|
+
+      timeline.push([
+        "#{task.id}", 
+        "#{task.type[0..-5]} #{task.equipment.type.titlecase} #{task.equipment.rhizome_eid}", 
```

Comment:
Instead of using #equipment, use #sprout. Otherwise, if the task is assigned to a Thermostat or RIMS #equipment will return nil. As-is, this will break the Schedules index page.